lib/phlex/renderable.rb

```diff
@@ -26,7 +26,9 @@ def render_in(view_context, &block)
 						output = yield(*args, **kwargs)
 						unchanged = (original_length == @_target.length)
 
-						text(output) if unchanged && output.is_a?(String)
+						if unchanged && output.is_a?(String)
+							output.html_safe? ? raw(output) : text(output)
+						end
```

Comment:
Calling `html_safe` will be unacceptably slow, but I’m happy to use a case statement to check for the `ActiveSupport::SafeBuffer` type. We just need to figure out how to handle `active_support` not being around, since it's not a dependency of Phlex.

```suggestion
						if unchanged
						  case output
						  when ActiveSupport::SafeBuffer
						  	raw(output)
						  when String
						  	text(output)
						  end
						end
```